server/src/main/java/org/elasticsearch/search/aggregations/bucket/terms/TermsAggregatorFactory.java

```diff
@@ -133,7 +134,7 @@ protected Aggregator doCreateInternal(ValuesSource valuesSource, Aggregator pare
             if (valuesSource instanceof ValuesSource.Bytes.WithOrdinals == false) {
                 execution = ExecutionMode.MAP;
             }
-            final long maxOrd = getMaxOrd(valuesSource, context.searcher());
+            final long maxOrd = getMaxOrd(context.searcher(), valuesSource, execution);
```

Comment:
it feels weird that the maximum ord depends on the execution mode
could we do something like `maxOrd = execution == MAP ? -1 : getMaxOrd(valuesSource, context.searcher());` instead? Existing code seems to be able to deal with -1 already because of fields that don't have ordinals at all like numbers.

Reply:
I opened https://github.com/elastic/elasticsearch/pull/38169 to address this